src/net/p2p.rs

```diff
@@ -2217,7 +2217,9 @@ impl PeerNetwork {
     /// -- runs the p2p and http peer main loop
     /// Returns the table of unhandled network messages to be acted upon, keyed by the neighbors
     /// that sent them (i.e. keyed by their event IDs)
-    pub fn run(&mut self, burndb: &BurnDB, chainstate: &mut StacksChainState, mempool: &mut MemPoolDB, dns_client_opt: Option<&mut DNSClient>, download_backpressure: bool, poll_timeout: u64) -> Result<NetworkResult, net_error> {
+    pub fn run(&mut self, burndb: &BurnDB, chainstate: &mut StacksChainState, mempool: &mut MemPoolDB,
```

Comment:
Can you put `exit_at_block_height` inside the `StacksChainState` struct?  There's gonna be a million things we'll want to feed into the RPC API at some point, so we should start thinking about ways to organize them better than passing a million arguments.

Reply:
I don't think it makes sense to add that to the StacksChainState struct.

What about introducing an RPCHandlerArgs struct for handling these kinds of additional args?

Reply:
Sure, having an explicit struct for runtime configuration state is fine.

Reply:
Addressed in c51a4e4cb